mempool/src/shared_mempool/peer_manager.rs

```diff
@@ -349,6 +359,30 @@ impl PeerManager {
             })
             .into_group_map();
 
+        // update default_failovers
+        // NOTE: this block of code, and maintaining even this concept of `default_failovers`, is a
+        // *temporary* patch to improve the reliability of txn delivery in case of the primary
+        // upstream network is lagging behind, and the txns delivered to it will not be ready for further
+        // broadcast/consensus on that end based on its stale state.
+        // So:
+        // (1) don't invest too much in refactoring this code w.r.t. the rest of this function
+        // (2) the logic of this function should later be migrated to a smarter networking layer
+        // that can handle peer selection for mempool
+        let mut default_failovers = self.default_failovers.lock();
+        if default_failovers.len() < self.mempool_config.default_failovers {
+            for failover_network in self.upstream_config.networks[1..].iter() {
+                if let Some(active_peers) = active_peers_by_network.get(failover_network) {
+                    for p in active_peers.iter().cloned() {
```

Comment:
Redundant clone?

Reply:
apparently this is a double reference, so it complains when I do a double clone inside the loop... :/